[DOCS-6531] MwL Tag Exclusion Updates (#20320)
DataDog/documentation: content/en/metrics/metrics-without-limits.md

Metrics without Limits™ provides you flexibility and control over your custom metrics volumes by decoupling custom metric ingestion and indexing. You only pay for custom metric tags that are valuable to your organization.

Metrics without Limits™ provides you with the ability to configure tags on all metric types in-app. You can also customize aggregations on counts, rates, and gauges without having to re-deploy or change any code. With Metrics without Limits™, you can configure an allowlist of tags in-app to remain queryable throughout the Datadog platform. This automatically excludes nonessential tags attached to application-level or business metrics (for example, `host`). Additionally with Metrics without Limits™, you can configure a blocklist of tags to quickly drop and exclude tags that provide less business value to you. This functionality is located in the [Metrics Summary][1] page.

This page identifies key components of Metrics without Limits™ that can help you manage your custom metrics volumes within your observability budget.

> ### Configuration of tags

Reviewer: ```suggestion
## Configuration of tags
```

> #### Configure an allowlist of tags 

Reviewer: ```suggestion
### Allowlist of tags 
```

> Click on any metric name to open its details sidepanel. Then click **Manage Tags** -> **"Include Tags..."** to configure the tags you'd like to remain as queryable on dashboards and monitors. The tag configuration modal pre-populates by default with an allowlist of tags that have been actively queried on dashboards, notebooks, monitors, and through API in the past 30 days (represented by an icon). You can also include your own additional tags. Before selecting **Save**, an estimated new volume of indexed custom metrics that results from this potential tag configuration is displayed.

Reviewer: ```suggestion
1. Click any metric name to open its details side panel. 
2. Click **Manage Tags** -> **"Include Tags..."** to configure the tags you'd like to remain as queryable on dashboards and monitors. 
3. Review or add tags in the configuration modal. By default, the tag configuration modal pre-populates an allowlist of tags that were actively queried on dashboards, notebooks, monitors, and through the API in the past 30 days (represented by an icon). 
4. Review the *Est. New Volume* of indexed custom metrics that results from this potential tag configuration.
4. Click **Save**.
```

{{< img src="metrics/mwl_example_include_tags.mp4" alt="Configuration of Tags with Allowlist" video=true >}}

> #### Configure an blocklist of tags 

Reviewer: ```suggestion
### Blocklist of tags 
```

> Click on any metric name to open its details sidepanel. Then click **Manage Tags** -> **“Exclude Tags…”** to drop tags you don't want to query for to optimize metric volumes. These tags will **not** remain queryable on dashboards and monitors. The tag configuration modal will let you know if you are selecting a tag that has been actively queried on dashboards, notebooks, monitors, and through API in the past 30 days (represented by an icon). Before selecting Save, an estimated new volume of indexed custom metrics that results from the potential tag configuration is displayed. 

Reviewer: Is there a still an icon that shows if a tag has been queried? I wasn't sure.
```suggestion
1. Click on any metric name to open its details side panel.
2. Click **Manage Tags** -> **"Exclude Tags…"** to drop tags you don't want to query. 
3. Review or add tags in the configuration modal. These tags are **not** queryable on dashboards and monitors.
4. Hover over a tag to see if it has been actively queried on dashboards, notebooks, monitors, and through the API in the past 30 days. 
5. Review the *Est. New Volume* of indexed custom metrics that results from the tag configuration. 
4. Click **Save**.
```

This page identifies key components of Metrics without Limits™ that can help you manage your custom metrics volumes within your observability budget.

## Configuration of tags

### Allowlist of tags 
1. Click any metric name to open its details side panel. 
2. Click **Manage Tags** -> **"Include Tags..."** to configure the tags you'd like to remain as queryable on dashboards and monitors. 
3. Review or add tags in the configuration modal. By default, the tag configuration modal pre-populates an allowlist of tags that were actively queried on dashboards, notebooks, monitors, and through the API in the past 30 days (represented by an icon). 

> 4. Review the *Est. New Volume* of indexed custom metrics that results from this potential tag configuration.

Reviewer 2: ```suggestion
4. Review the *Estimated New Volume* of indexed custom metrics that results from this potential tag configuration.
```

4. Click **Save**.

{{< img src="metrics/mwl_example_include_tags.mp4" alt="Configuration of Tags with Allowlist" video=true >}}

### Blocklist of tags 
1. Click on any metric name to open its details side panel.
2. Click **Manage Tags** -> **"Exclude Tags…"** to drop tags you don't want to query. 

> 3. Review or add tags in the configuration modal. These tags are **not** queryable on dashboards and monitors.

Reviewer 2: ```suggestion
3. Define your blocklist of tags. These tags will **not** be queryable on dashboards and monitors.
```

> 4. Hover over a tag to see if it has been actively queried on dashboards, notebooks, monitors, and through the API in the past 30 days. 

Reviewer 2: Merge steps 3/4 so we have equal number of steps for inclusion/exclusion

Reviewer: I think it's ok to have more steps for one over the other. Having bigger blocks of information can be harder to digest.

Author: Since steps 3 and 4 talk about looking for queryable tags in the configuration modal, it would make sense to combine the two together

> 5. Review the *Est. New Volume* of indexed custom metrics that results from the tag configuration. 

Reviewer 2: ```suggestion
4. Review the *Estimated New Volume* of indexed custom metrics that results from this potential tag configuration.
```

Metrics without Limits™ provides you flexibility and control over your custom metrics volumes by decoupling custom metric ingestion and indexing. You only pay for custom metric tags that are valuable to your organization.

Metrics without Limits™ provides you with the ability to configure tags on all metric types in-app. You can also customize aggregations on counts, rates, and gauges without having to re-deploy or change any code. With Metrics without Limits™, you can configure an allowlist of tags in-app to remain queryable throughout the Datadog platform; this automatically drops nonessential tags attached to application-level or business metrics (for example, `host`). Alternatively, you can configure a blocklist of tags in-app to quickly drop and exclude tags; this automatically retains remaining essential tags that provide business value to your teams. These configuration functionalities are located in the [Metrics Summary][1] page.

This page identifies key components of Metrics without Limits™ that can help you manage your custom metrics volumes within your observability budget.

> ### Configuration of tags

Reviewer: @ijkaylin @ouracheal Is there a reason you want this to be a nested header under ## Overview rather than ## Configuration of tags
- From an information standpoint, it seems like it's a separate section and moving it out will also make sure the header is listed in the Table of contents on the right.
- Also, I recommend moving this section down below the Getting Started, unless you're seeing a lot of customer questions about this.

> #### Allowlist of tags 

Reviewer: Recommend removing redundant reference to tags.
```suggestion
#### Allowlist 
```

Author: Hi @estherk15! I see that Allowlist is under Configuration of tags so it makes sense to remove the "of tags" part of the header. Thanks!

1. Click any metric name to open its details side panel. 

> 2. Click **Manage Tags** -> **"Include Tags..."** to configure the tags you'd like to remain as queryable throughout Datadog (i.e. Dashboards, Notebooks, Monitors etc.)

Reviewer: ```suggestion
2. Click **Manage Tags** -> **"Include Tags..."** to configure the tags you'd like to remain as queryable in dashboards, notebooks, monitors, and other Datadog products.
```

3. Define your allowlist of tags. 

> By default, the tag configuration modal pre-populates with a Datadog recommended allowlist of tags that have been actively queried on dashboards, notebooks, monitors, or via the API in the past 30 days. Our recommendations are distinguished with an icon. 

Reviewer: What is the icon?

> By default, the tag configuration modal pre-populates with a Datadog recommended allowlist of tags that have been actively queried on dashboards, notebooks, monitors, or via the API in the past 30 days. Our recommendations are distinguished with an icon. 

Author: The icon looks like [this](https://a.cl.ly/L1u8XkE6)

4. Review the *Estimated New Volume* of indexed custom metrics that results from this potential tag configuration.

> 4. Click **Save**.

Reviewer: ```suggestion
5. Click **Save**.
```

{{< img src="metrics/mwl_example_include_tags.mp4" alt="Configuration of Tags with Allowlist" video=true >}}

### Blocklist of tags 
1. Click on any metric name to open its details side panel.
2. Click **Manage Tags** -> **"Exclude Tags…"** to drop tags you don't want to query. 
3. Define your blocklist of tags. These tags will **not** be queryable on dashboards and monitors.
4. Hover over a tag to see if it has been actively queried on dashboards, notebooks, monitors, and through the API in the past 30 days. 

> 4. Review the *Estimated New Volume* of indexed custom metrics that results from this potential tag configuration.

Reviewer: ```suggestion
5. Review the *Estimated New Volume* of indexed custom metrics that results from this potential tag configuration.
```

> 4. Click **Save**.

Reviewer: ```suggestion
6. Click **Save**.
```

This page identifies key components of Metrics without Limits™ that can help you manage your custom metrics volumes within your observability budget.

### Configuration of tags

#### Allowlist of tags 
1. Click any metric name to open its details side panel. 
2. Click **Manage Tags** -> **"Include Tags..."** to configure the tags you'd like to remain as queryable in dashboards, notebooks, monitors, and other Datadog products.
3. Define your allowlist of tags. 
By default, the tag configuration modal pre-populates with a Datadog recommended allowlist of tags that have been actively queried on dashboards, notebooks, monitors, or via the API in the past 30 days. Our recommendations are distinguished with an icon. 
4. Review the *Estimated New Volume* of indexed custom metrics that results from this potential tag configuration.
5. Click **Save**.

{{< img src="metrics/mwl_example_include_tags-compressed.mp4" alt="Configuration of Tags with Allowlist" video=true style="width:100%" >}}

> There are APIs available to [create][2], [edit][3], and [delete][4] a tag configuration along with an [API][5] for estimating the potential impact of your configuration.

Reviewer: ```suggestion
You can [create][2], [edit][3], [delete][4], and [estimate the impact][5] of your tag configuration through the Metrics APIs.
```

#### Blocklist of tags 
1. Click on any metric name to open its details side panel.
2. Click **Manage Tags** -> **"Exclude Tags…"** to drop tags you don't want to query. 

> 3. Define your blocklist of tags. These tags will **not** be queryable on dashboards and monitors. Tags that have been actively queried on dashboards, notebooks, monitors, and through the API in the past 30 days will be distinguished with an icon.

Reviewer: ```suggestion
3. Define your blocklist of tags. These tags are **not** queryable on dashboards and monitors. Tags that have been actively queried on dashboards, notebooks, monitors, and through the API in the past 30 days will be distinguished with a graph line icon.
```

Author: I think it makes more sense here to say these tags "will not be queryable" rather than "these tags are not queryable" as we are talking about setting the blocklist, not the list being set already.

Reviewer: How about this? We try to avoid future tense in docs
```suggestion
3. Define your blocklist of tags. Tags defined in the blocklist are **not** queryable on dashboards and monitors. Tags that have been actively queried on dashboards, notebooks, monitors, and through the API in the past 30 days are distinguished with a graph line icon.
```

5. Review the *Estimated New Volume* of indexed custom metrics that results from this potential tag configuration.
6. Click **Save**.

{{< img src="metrics/mwl_example_tag_exclusion-compressed.mp4" alt="Configuration of Tags with Tag Exclusion" video=true style="width:100%" >}}

> Similar to the allowlist of tags, you can [create][2] and [edit][3] a blocklist of tags by setting the parameter `exclude_tags_mode: true`.

Reviewer: ```suggestion
Set the parameter `exclude_tags_mode: true` on the Metrics API to [create][2] and [edit][3] a blocklist of tags.
```

When configuring tags for counts, rates, and gauges, the most frequently queried time/space aggregation combination is available for query by default.

### Configure multiple metrics at a time

Optimize your custom metrics volumes by using the [bulk metric tag configuration feature][7]. To specify a namespace for your metrics, click **Configure Tags*** on Metrics Summary. You can configure all metrics matching that namespace prefix with the same allowlist of tags under ***Include tags...*** or the same blocklist of tags under ***Exclude tags...***.

> This is also available via API so you can [configure][13] and [delete][14] tags for multiple metrics. To configure a blocklist of tags for multiple metrics, set the parameter `exclude_tags_mode: true` using the [configuration][13] API.

Reviewer: ```suggestion
You can [configure][13] and [delete][14] tags for multiple metrics through the API. To [configure a blocklist of tags][13] for multiple metrics, set the parameter `exclude_tags_mode: true` on the API.
```

### Configure multiple metrics at a time

> Optimize your custom metrics volumes by using the [bulk metric tag configuration feature][7]. By clicking **Configure Tags*** on Metrics Summary, you can specify a namespace for your metrics. You can then configure all metrics matching that namespace prefix with the same allowlist of tags under ***Include tags...*** or the same blocklist of tags under ***Exclude tags...***.

Reviewer: ```suggestion
Optimize your custom metrics volumes by using the [bulk metric tag configuration feature][7]. Click **Configure Tags*** on Metrics Summary, to specify a namespace for your metrics. You can configure all metrics matching that namespace prefix with the same allowlist of tags under ***Include tags...*** or the same blocklist of tags under ***Exclude tags...***.
```

### Configure multiple metrics at a time

> Optimize your custom metrics volumes by using the [bulk metric tag configuration feature][7]. Click **Configure Tags*** on Metrics Summary, to specify a namespace for your metrics. You can configure all metrics matching that namespace prefix with the same allowlist of tags under ***Include tags...*** or the same blocklist of tags under ***Exclude tags...***.

Reviewer: ```suggestion
Optimize your custom metrics volumes by using the [bulk metric tag configuration feature][7]. To specify a namespace for your metrics, click **Configure Tags*** on the Metrics Summary page. You can configure all metrics matching that namespace prefix with the same allowlist of tags under ***Include tags...*** or the same blocklist of tags under ***Exclude tags...***.
```

> Optimize your custom metrics volumes by using the [bulk metric tag configuration feature][7]. To specify a namespace for your metrics, click **Configure Tags*** on the Metrics Summary page. You can configure all metrics matching that namespace prefix with the same allowlist of tags under ***Include tags...*** or the same blocklist of tags under ***Exclude tags...***.

Reviewer 2: ```suggestion
Optimize your custom metrics volumes by using the [bulk metric tag configuration feature][7]. To specify a namespace for your metrics, click **Configure Tags*** on Metrics Summary. You can configure all metrics matching that namespace prefix with the same allowlist of tags under ***Include tags...*** or the same blocklist of tags under ***Exclude tags...***.
```
